CMakeLists.txt

```
@@ -597,21 +596,32 @@ message("Configure summary")
 message("=================")
 message("Executables:")
 message("  bitcoind ............................ ${BUILD_DAEMON}")
-message("  bitcoin-qt .......................... ${BUILD_GUI}")
+message("  bitcoin-node (multiprocess) ......... ${BUILD_MULTIPROCESS}")
+message("  bitcoin-qt (GUI) .................... ${BUILD_GUI}")
+if(BUILD_GUI AND BUILD_MULTIPROCESS)
+  set(bitcoin_gui_status ON)
+else()
+  set(bitcoin_gui_status OFF)
+endif()
+message("  bitcoin-gui (GUI, multiprocess) ..... ${bitcoin_gui_status}")
 message("  bitcoin-cli ......................... ${BUILD_CLI}")
 message("  bitcoin-tx .......................... ${BUILD_TX}")
 message("  bitcoin-util ........................ ${BUILD_UTIL}")
 message("  bitcoin-wallet ...................... ${BUILD_WALLET_TOOL}")
 message("  bitcoin-chainstate (experimental) ... ${BUILD_UTIL_CHAINSTATE}")
 message("  libbitcoinkernel (experimental) ..... ${BUILD_KERNEL_LIB}")
-message("Wallet support:")
-message("  SQLite, descriptor wallets .......... ${WITH_SQLITE}")
-message("  Berkeley DB, legacy wallets ......... ${WITH_BDB}")
-message("Optional packages:")
-message("  external signer ..................... ${WITH_EXTERNAL_SIGNER}")
-message("  multiprocess ........................ ${MULTIPROCESS}")
-message("  NAT-PMP ............................. ${WITH_NATPMP}")
-message("  UPnP ................................ ${WITH_MINIUPNPC}")
+message("Optional features:")
+message("  wallet support ...................... ${ENABLE_WALLET}")
+if(ENABLE_WALLET)
+  message("   - descriptor wallets (SQLite) ...... ${WITH_SQLITE}")
+  message("   - legacy wallets (Berkeley DB) ..... ${WITH_BDB}")
+endif()
+message("  external signer ..................... ${ENABLE_EXTERNAL_SIGNER}")
+message("  port mapping:")
+if(WITH_NATPMP OR WITH_MINIUPNPC)
+  message("   - using NAT-PMP .................... ${WITH_NATPMP}")
+  message("   - using UPnP ....................... ${WITH_MINIUPNPC}")
+endif()
```

non-blocker: I would rather have a disabled option being present in the summary with an `OFF` value instead of missing. Also this looks inconsistent:

```
Optional features:
  wallet support ...................... ON
   - descriptor wallets (SQLite) ...... ON
   - legacy wallets (Berkeley DB) ..... OFF
  external signer ..................... ON
  port mapping:                          <-- no ....dots and no ON or OFF on this line, has nothing with deeper indentation underneath it either
  ZeroMQ .............................. OFF
  USDT tracing ........................ OFF
  QR code (GUI) ....................... OFF
  DBus (GUI, Linux only) .............. OFF
```

```suggestion
message("  port mapping:")
message("   - using NAT-PMP .................... ${WITH_NATPMP}")
message("   - using UPnP ....................... ${WITH_MINIUPNPC}")
```

---

I agree, or `no port mapping options enabled (NAT-PMP, UPnP)` (?), but I think it'd be better to show features in `OFF`.

---

Thanks! [Reworked](https://github.com/hebasto/bitcoin/pull/198#issuecomment-2122259823).